components/service-operator/internal/aws/cloudformation/controller.go

```diff
@@ -227,9 +227,52 @@ func (r *Controller) reconcileObjectWithContext(ctx context.Context, req ctrl.Re
 		}
 	}
 
+	if stackThatWritesServiceEntry, ok := o.(ServiceEntryCreator); ok {
+		err = r.updateServiceEntry(ctx, stackThatWritesServiceEntry, outputs)
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// updateServiceEntry will write any necessary Istio ServiceEntry resource to
+// Kubernetes, so that Istio will allow the tenant's pods to egress to the
+// resource we've created for them.
+func (r *Controller) updateServiceEntry(ctx context.Context, o ServiceEntryCreator, outputs Outputs) error {
+	serviceEntry, err := o.GetServiceEntry(outputs)
+	if err != nil {
+		return err
+	}
+
+	serviceEntryKey, err := client.ObjectKeyFromObject(serviceEntry)
+	if err != nil {
+		return err
+	}
+	err = r.KubernetesClient.Get(ctx, serviceEntryKey, serviceEntry)
+	if err != nil && !apierrs.IsNotFound(err) {
+		return err
+	}
+	op, err := controllerutil.CreateOrUpdate(ctx, r.KubernetesClient, serviceEntry, func() error {
+		return nil
+	})
```

Comment:
The serviceEntry object should be mutated inside this `func` ... for example it should be something like:

```
op, err := controllerutil.CreateOrUpdate(ctx, r.KubernetesClient, serviceEntry, func() error {
    serviceEntry.Spec = o.GetServiceEntry().Spec
    return nil
})
```